src/v/kafka/server/handlers/details/security.h

```diff
@@ -159,6 +160,15 @@ inline security::acl_binding to_acl_binding(const creatable_acl& acl) {
           fmt::format("Invalid cluster name: {}", pattern.name()));
     }
 
+    if (pattern.resource() == security::resource_type::topic) {
+        if (pattern.name() != "*" && model::validate_kafka_topic_name(
+              model::topic_view(pattern.name()))) {
+            throw acl_conversion_error(fmt::format(
+              "ACL topic does not conform to kafka topic schema: {}",
+              pattern.name()));
```

Comment:
suggestion: May be helpful to return why (from the error code returned by `validate_kafka_topic_name`) the validation failed.

Reply:
At least maybe in the error message thrown here

Reply:
sounds good

Reply:
+1